### ISSUE TITLE
Ruffle Crashes When playing Freeing the Complex - Cannot recursively aquire mutex

### ISSUE THREAD

**Issue by nailuj29** [ruffle-rs/ruffle#1931]
![image](https://user-images.githubusercontent.com/54189157/101940803-0ab9cc00-3bb5-11eb-9912-297ca46b3608.png)
Error message:
```
panicked at 'cannot recursively acquire mutex', /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys/wasm/../unsupported/mutex.rs:22:9

Stack:

Q@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/core.ruffle.js:1:20738
__wbg_new_59cb74e423758ede@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/ruffle.js:1:35075
@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/16f1c86940c88c5ecffb.module.wasm:wasm-function[1737]:0x2779de
@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/16f1c86940c88c5ecffb.module.wasm:wasm-function[1056]:0x206a03
@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/16f1c86940c88c5ecffb.module.wasm:wasm-function[1379]:0x2452e4
@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/16f1c86940c88c5ecffb.module.wasm:wasm-function[3696]:0x2f9071
@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1daee31/dist/16f1c86940c88c5ecffb.module.wasm:wasm-function[2923]:0x2db9ea
@moz-extension://2047b1f3-3491-48b5-a6ba-79bbc1dae…
```
SWF Link: [https://uploads.ungrounded.net/665000/665992_complex.swf?1447350089](https://uploads.ungrounded.net/665000/665992_complex.swf?1447350089)

Version: 12/8/2020 nightly build

**Comment by nailuj29gaming** [comment]
This also happens on every other henry stickmin game

**Comment by jhwgh1968** [comment]
I tried this with the desktop client with 310524920b2 (current master), and I got a blank screen with this error spammed infinitely:

```
[2020-12-16Txx:xx:xxZ ERROR ruffle_render_wgpu] Tessellation failure: TooManyVertices
```

I tried repeatedly to add debug prints or improve the log statements, but nothing ever changed. Rendering trace information was completely unhelpful as well.

I cannot figure out where it is coming from, except somewhere deep in the Lyon library.
